backend/src/test/java/gov/cdc/usds/simplereport/service/TestOrderServiceTest.java

```diff
@@ -541,6 +542,45 @@ void addTestResult_testAlreadySubmitted_failure() {
                 devA.getInternalId(), TestResult.POSITIVE, p.getInternalId(), null));
   }
 
+  @Test
+  @WithSimpleReportOrgAdminUser
+  void addTestResult_correction() {
+    Organization org = _organizationService.getCurrentOrganization();
+    Facility facility = _organizationService.getFacilities(org).get(0);
+    Person p = _dataFactory.createFullPerson(org);
+    _service.addPatientToQueue(
+        facility.getInternalId(), p, "", Collections.emptyMap(), LocalDate.of(1865, 12, 25), false);
+    DeviceSpecimenType devA = _dataFactory.getGenericDeviceSpecimen();
+    facility.addDefaultDeviceSpecimen(devA);
+
+    // Create test event for a later correction
+    _service.addTestResult(devA.getInternalId(), TestResult.POSITIVE, p.getInternalId(), null);
+    List<TestEvent> testEvents =
+        _testEventRepository.findAllByPatientAndFacilities(p, List.of(facility));
+    assertEquals(1, testEvents.size());
+
+    // Mark existing test order as "corrected", re-open as "pending" order status
+    TestEvent originalTestEvent = testEvents.get(0);
+    _service.markAsCorrection(originalTestEvent.getInternalId(), "Cold feet");
+
+    // Issue test correction
+    _service.addTestResult(devA.getInternalId(), TestResult.NEGATIVE, p.getInternalId(), null);
+
+    // Get newly-created correction event
+    TestEvent correctionTestEvent =
+        _testEventRepository.findAllByPatientAndFacilities(p, List.of(facility)).stream()
+            .filter(event -> event.getCorrectionStatus() == TestCorrectionStatus.CORRECTED)
+            .collect(Collectors.toList())
+            .get(0);
+
+    assertEquals(
+        originalTestEvent.getInternalId(), correctionTestEvent.getPriorCorrectedTestEventId());
+    assertEquals(TestCorrectionStatus.CORRECTED, correctionTestEvent.getCorrectionStatus());
+    assertEquals("Cold feet", correctionTestEvent.getReasonForCorrection());
+    assertEquals(TestResult.NEGATIVE, correctionTestEvent.getResult());
+    // think about the dates here - how's backdating look?
```

Comment:
Not sure if this comment is still meant to be here?

Reply:
Good eye. Updated in [7f3504e](https://github.com/CDCgov/prime-simplereport/pull/3616/commits/7f3504ede3ff1f6d1bec24d86b4f0ed0f84908d3).